packages/nextjs/src/utils/instrumentServer.ts

```diff
@@ -173,35 +174,43 @@ function makeWrappedReqHandler(origReqHandler: ReqHandler): WrappedReqHandler {
     res: NextResponse,
     parsedUrl?: url.UrlWithParsedQuery,
   ): Promise<void> {
-    // We only want to record page and API requests
-    if (hasTracingEnabled() && shouldTraceRequest(req.url, publicDirFiles)) {
-      const transaction = Sentry.startTransaction({
-        name: `${(req.method || 'GET').toUpperCase()} ${req.url}`,
-        op: 'http.server',
-      });
-      Sentry.getCurrentHub()
-        .getScope()
-        ?.setSpan(transaction);
-
-      res.__sentry__ = {};
-      res.__sentry__.transaction = transaction;
-    }
-
-    res.once('finish', () => {
-      const transaction = res.__sentry__?.transaction;
-      if (transaction) {
-        // Push `transaction.finish` to the next event loop so open spans have a chance to finish before the transaction
-        // closes
-        setImmediate(() => {
-          // TODO
-          // addExpressReqToTransaction(transaction, req);
-          transaction.setHttpStatus(res.statusCode);
-          transaction.finish();
+    // wrap everything in a domain in order to prevent scope bleed between requests
+    const local = domain.create();
+    local.add(req);
+    local.add(res);
+    // TODO could this replace wrapping the error logger?
+    // local.on('error', Sentry.captureException);
+
+    local.run(() => {
+      // We only want to record page and API requests
+      if (hasTracingEnabled() && shouldTraceRequest(req.url, publicDirFiles)) {
+        const transaction = Sentry.startTransaction({
+          name: `${(req.method || 'GET').toUpperCase()} ${req.url}`,
```

Comment:
I think we don't usually do this with transaction names (ex. `GET route/to/entity`), this is a pattern we use for http spans names. 

Looking at all our api routes in Sentry, we usually just use the URL. We can add the method as a tag. Also, is it fine to assume that if `req.method` not set, it'll always be a `GET`?

Reply:
After IRL conversation, we decided to leave it for API requests (which actually might be `PUT` or `POST`) but take it off for regular route/pageload transactions.